backend/backend.go

```diff
@@ -822,17 +816,18 @@ func (backend *Backend) uninitAccounts() {
 	backend.accounts = []accounts.Interface{}
 }
 
-// Keystores returns the keystores registered at this backend.
-func (backend *Backend) Keystores() *keystore.Keystores {
-	return backend.keystores
+// Keystore returns the keystore registered at this backend, or nil if no keystore is registered.
+func (backend *Backend) Keystore() keystore.Keystore {
+	defer backend.accountsAndKeystoreLock.RLock()()
+	return backend.keystore
 }
 
 // registerKeystore registers the given keystore at this backend.
 func (backend *Backend) registerKeystore(keystore keystore.Keystore) {
+	defer backend.accountsAndKeystoreLock.Lock()()
+
 	backend.log.Info("registering keystore")
```

Comment:
how about adding keystore.RootFingerprint() to the message? might help with debugging? and the same in "deregestering keystore".

Reply:
Good idea, done